// @mantium/mantiumapi/lib/config/config.js
"use strict";

var index = require('../index');

module.exports = {
  /*
   * Construct base URL
   */
  baseURL: function baseURL() {
    return "".concat(index.getOrigin(), "/").concat(index.getApiVersion());
  },

  /*
   * Auth endpoints
   * https://developer.mantiumai.com/reference#auth
   */
  accessTokenLoginURL: function accessTokenLoginURL() {
    return "".concat(this.baseURL(), "/auth/login/access/token");
  },
  revokeTokenURL: function revokeTokenURL() {
    return "".concat(this.baseURL(), "/auth/user/revoke/token");
  },

  /*
   * List Methods
   * https://developer.mantiumai.com/reference#ai_methods
   */
  aiMethodsURL: function aiMethodsURL() {
    return "".concat(this.baseURL(), "/ai_methods");
  },

  /*
   * Get All Ai Engines
   * https://developer.mantiumai.com/reference#ai-engines
   */
  aiEnginesAllURL: function aiEnginesAllURL() {
    return "".concat(this.baseURL(), "/ai/engine/all");
  },
  aiEnginesByProviderURL: function aiEnginesByProviderURL() {
    return "".concat(this.baseURL(), "/ai/engine/get/ai/providers");
  },
  aiEnginesByNameURL: function aiEnginesByNameURL() {
    return "".concat(this.baseURL(), "/ai/engine/get/name");
  },

  /*
   * Tags
   * https://developer.mantiumai.com/reference#tags
   */
  tagsURL: function tagsURL() {
    var isIdURL = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    return "".concat(this.baseURL(), "/tag").concat(isIdURL ? '/id' : '');
  },

  /*
   * Prompts
   * https://developer.mantiumai.com/reference#prompts
   */
  promptURL: function promptURL() {
    var isIdURL = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    return "".concat(this.baseURL(), "/prompt").concat(isIdURL ? '/id' : '');
  },
  promptExecuteURL: function promptExecuteURL(id) {
    return "".concat(this.baseURL(), "/prompt/").concat(id, "/execute");
  },
  promptResultURL: function promptResultURL() {
    return "".concat(this.baseURL(), "/prompt/result");
  },
  promptTryURL: function promptTryURL(id) {
    return "".concat(this.baseURL(), "/prompt/").concat(id, "/try");
  },

  /*
   * Logs
   * https://developer.mantiumai.com/reference#logs
   */
  logsURL: function logsURL() {
    var isIdURL = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    return "".concat(this.baseURL(), "/log").concat(isIdURL ? '/id' : '');
  },

  /*
   * Intelet
   * https://developer.mantiumai.com/reference#intelets
   */
  inteletURL: function inteletURL() {
    var isIdURL = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    return "".concat(this.baseURL(), "/intelet").concat(isIdURL ? '/id' : '');
  },
  inteletExecuteURL: function inteletExecuteURL(id) {
    return "".concat(this.baseURL(), "/intelet/").concat(id, "/execute");
  },
  inteletResultURL: function inteletResultURL() {
    return "".concat(this.baseURL(), "/intelet/result");
  },

  /*
   * API Health
   * https://developer.mantiumai.com/reference/api_health_health_get
   */
  apiHealthURL: function apiHealthURL() {
    return "".concat(index.getOrigin(), "/health");
  },

  /*
   * API Files
   * https://developer.mantiumai.com/reference/get_files_v1_files_openai_files_get
   */
  filesURL: function filesURL() {
    return "".concat(this.baseURL(), "/files/openai_files");
  },
  filesUploadURL: function filesUploadURL() {
    return "".concat(this.baseURL(), "/files/aws/signature");
  },
  filesFinetuneURL: function filesFinetuneURL() {
    return "".concat(this.baseURL(), "/files/finetunes");
  },

  /*
   * Security
   * https://developer.mantiumai.com/reference/post_policy_v1_security_policy_post
   */
  securityURL: function securityURL() {
    var isIdURL = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
    var subUrl = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : null;
    return "".concat(this.baseURL(), "/security").concat(subUrl ? '/' + subUrl : '').concat(isIdURL ? '/id' : '');
  }
};
